src/krislet/Actions.java

```diff
@@ -10,8 +10,16 @@ public FindAction(String target) {
             this.target = target;
         }
 
-        public void excute(Krislet krislet) {
-            krislet.turn(40);
+        public void execute(Krislet krislet) {
+        	// Check if target is in view
+        	ObjectInfo object;
+        	object = krislet.getMemory().getObject(this.target);
+        	// If target is not in view, turn
+        	if (object == null)
+        	{
+        		krislet.turn(40);
+        		krislet.getMemory().waitForNewInfo();
```

Comment:
So I don't think we want to wait for new objects here, since this is handled in the searchForPercets function.